src/shogun/kernel/Kernel.h

```diff
@@ -518,7 +518,7 @@ class Kernel : public SGObject
 		 */
 		virtual int32_t get_num_vec_rhs()
 		{
-			return num_rhs;
+			return rhs->get_num_vectors();
```

Comment:
Can you add something like this please? For both examples. Feel free to improve the error message
```suggestion
			require(rhs, "Kernel::rhs has not been set");
			return rhs->get_num_vectors();
```

Reply:
Actually, the error message should read "Kernel right hand side features not set"

(User facing, to avoid putting variable/member names in error messages, but instead put the semantic equivalent)